### PR TITLE
[SPARK-31087][SQL] Add Back Multiple Removed APIs 

### PR DESCRIPTION
**Pull request by gatorsmile**
### What changes were proposed in this pull request?

Based on the discussion in the mailing list [[Proposal] Modification to Spark's Semantic Versioning Policy](http://apache-spark-developers-list.1001551.n3.nabble.com/Proposal-Modification-to-Spark-s-Semantic-Versioning-Policy-td28938.html) , this PR is to add back the following APIs whose maintenance cost are relatively small.

- functions.toDegrees/toRadians 
- functions.approxCountDistinct 
- functions.monotonicallyIncreasingId
- Column.!==
- Dataset.explode
- Dataset.registerTempTable
- SQLContext.getOrCreate, setActive, clearActive, constructors

Below is the other removed APIs in the original PR, but not added back in this PR [https://issues.apache.org/jira/browse/SPARK-25908]:

- Remove some AccumulableInfo .apply() methods
- Remove non-label-specific multiclass precision/recall/fScore in favor of accuracy
- Remove unused Python StorageLevel constants
- Remove unused multiclass option in libsvm parsing
- Remove references to deprecated spark configs like spark.yarn.am.port
- Remove TaskContext.isRunningLocally
- Remove ShuffleMetrics.shuffle* methods
- Remove BaseReadWrite.context in favor of session

### Why are the changes needed?
Avoid breaking the APIs that are commonly used.

### Does this PR introduce any user-facing change?
Adding back the APIs that were removed in 3.0 branch does not introduce the user-facing changes, because Spark 3.0 has not been released. 

### How was this patch tested?
Added a new test suite for these APIs.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Hi, @gatorsmile .
Can we have a new JIRA ID?

**Review by dongjoon-hyun — COMMENTED** [review]
Could you summarize what is not recovered? The original list is the following.
```
    - Remove some AccumulableInfo .apply() methods
    - Remove non-label-specific multiclass precision/recall/fScore in favor of accuracy
    - Remove toDegrees/toRadians in favor of degrees/radians (SparkR: only deprecated)
    - Remove approxCountDistinct in favor of approx_count_distinct (SparkR: only deprecated)
    - Remove unused Python StorageLevel constants
    - Remove Dataset unionAll in favor of union
    - Remove unused multiclass option in libsvm parsing
    - Remove references to deprecated spark configs like spark.yarn.am.port
    - Remove TaskContext.isRunningLocally
    - Remove ShuffleMetrics.shuffle* methods
    - Remove BaseReadWrite.context in favor of session
    - Remove Column.!== in favor of =!=
    - Remove Dataset.explode
    - Remove Dataset.registerTempTable
    - Remove SQLContext.getOrCreate, setActive, clearActive, constructors
```

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Please hold on until the vote.

**Comment by maropu** [comment]
btw, could we update the contribution guideline to use a new tag in a title for these kinds of user-facing changes, e.g., `[SPARK-XXX][SQL][API] XXX`? I personally think the tag could make us easily track on-going PRs and commit logs for high-impact changes.

**Comment by gatorsmile** [comment]
retest this please

**Comment by gatorsmile** [comment]
CC @dongjoon-hyun @marmbrus @rxin @srowen This PR is ready for review.

**Comment by gatorsmile** [comment]
retest this please

**Review by srowen — COMMENTED** [review]
I don't object to this. These are mainly aliases, which are easy to maintain. I also do not see any argument these are used though; it seems kind of arbitrary. What does it mean to deprecate these things and never remove them? @gatorsmile

**Review by dongjoon-hyun — APPROVED** [review]

**Review by cloud-fan — APPROVED** [review]

**Review by maropu — APPROVED** [review]

**Comment by gatorsmile** [comment]
Thanks! Merged to master/3.0

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by gatorsmile** [comment]
Retest this please.

**Comment by maropu** [comment]
retest this please

**Comment by gatorsmile** [comment]
retest this please

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by gatorsmile** [comment]
Thanks！ Merged to master/3.0
